util/chunk/row_container.go

```diff
@@ -28,149 +31,170 @@ import (
 
 // RowContainer provides a place for many rows, so many that we might want to spill them into disk.
 type RowContainer struct {
-	// records stores the chunks in memory.
-	records *List
-	// recordsInDisk stores the chunks in disk.
-	recordsInDisk *ListInDisk
+	m struct {
+		// RWMutex guarantees spill and get operator for rowContainer is mutually exclusive.
+		sync.RWMutex
+		// records stores the chunks in memory.
+		records *List
+		// recordsInDisk stores the chunks in disk.
+		recordsInDisk *ListInDisk
+		// spillError stores the error when spilling.
+		spillError error
+		// spilled indicate whether the RowContainer is spilled.
+		spilled uint32
```

Comment:
We need to add comment to explain that we need to use `atomic` when trying to modify `spilled`